src/pages/Home/components/Card/Card.jsx

```diff
@@ -1,29 +1,92 @@
-import React from 'react';
-import { arrayOf, number, string } from 'prop-types';
-
-import { Button } from 'src/components';
+import React, { useCallback, useState, useEffect, useRef } from 'react';
+import {
+  arrayOf,
+  number,
+  oneOfType,
+  string,
+  func,
+  instanceOf,
+} from 'prop-types';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faEllipsisV } from '@fortawesome/free-solid-svg-icons';
 
-export const Card = ({ url, title, genres, releaseDate }) => {
+import { Button, Menu } from 'src/components';
+
+export const Card = ({
+  imageUrl,
+  title,
+  genre,
+  releaseDate,
+  id,
+  onMenuSelect,
+  menuOptions,
+}) => {
+  const [isMenuOpen, setMenuOpen] = useState(false);
+
+  const menuRef = useRef(null);
+
+  const handleActionClick = useCallback(() => {
+    setMenuOpen(isOpen => !isOpen);
+  }, []);
+
+  const handleMenuItemSelect = useCallback(
+    action => {
+      onMenuSelect(id, action);
+      setMenuOpen(false);
+    },
+    [onMenuSelect, id],
+  );
+
+  const handleOutsideClick = useCallback(e => {
+    if (menuRef.current && !menuRef.current.contains(e.target)) {
+      setMenuOpen(false);
+    }
+  }, []);
+
+  useEffect(() => {
+    if (isMenuOpen) {
+      document.addEventListener('mousedown', handleOutsideClick);
+    } else {
+      document.removeEventListener('mousedown', handleOutsideClick);
+    }
+  }, [isMenuOpen, handleOutsideClick]);
```

Comment:
What if we move to another page? We will have a memory leak here. Use effect should remove listener additionally in the function returned from the `useEffect`.

Reply:
also fixed in next PR